app/components/user-menu.hbs

```diff
@@ -24,22 +24,23 @@
     <ClickOutside @onClickOutside={{fn (set this.isOpen) false}}>
       <ul class="menu">
         <li tabindex="-1" data-test-item>
-          <LinkTo @route="myprofile" @invokeAction={{set this.isOpen false}}>
+          <LinkToWithAction @route="myprofile" @action={{set this.isOpen false}} @queryParams={{hash invalidatetokens=null newtoken=null}}>
             {{t "general.myProfile"}}
-          </LinkTo>
+          </LinkToWithAction>
         </li>
         <li tabindex="-1" data-test-item>
-          <LinkTo
+          <LinkToWithAction
             @route="mymaterials"
-            @invokeAction={{set this.isOpen false}}
+            @action={{set this.isOpen false}}
+            @queryParams={{hash course=null filter=null sortBy=null firstOfferingDate=null}}
```

Comment:
this blows up when you set `sortBy` to `null` with a `TypeError: sortKey is null` error.

```suggestion
            @queryParams={{hash course=null filter=null firstOfferingDate=null}}
```

Reply:
This doesn't end up setting the default correctly as it populates it on the URL string. I'm gonna try changing this controller a bit in common to handle this. Good catch!